compactor/periodic.go

```diff
@@ -26,72 +26,60 @@ import (
 )
 
 // Periodic compacts the log by purging revisions older than
-// the configured retention time. Compaction happens hourly.
+// the configured retention time.
 type Periodic struct {
-	clock        clockwork.Clock
-	periodInHour int
+	clock  clockwork.Clock
+	period time.Duration
 
 	rg RevGetter
 	c  Compactable
 
-	revs   []int64
 	ctx    context.Context
 	cancel context.CancelFunc
 
-	mu     sync.Mutex
+	// mu protects paused
+	mu     sync.RWMutex
 	paused bool
 }
 
 // NewPeriodic creates a new instance of Periodic compactor that purges
-// the log older than h hours.
-func NewPeriodic(h int, rg RevGetter, c Compactable) *Periodic {
+// the log older than h Duration.
+func NewPeriodic(h time.Duration, rg RevGetter, c Compactable) *Periodic {
 	return &Periodic{
-		clock:        clockwork.NewRealClock(),
-		periodInHour: h,
-		rg:           rg,
-		c:            c,
+		clock:  clockwork.NewRealClock(),
+		period: h,
+		rg:     rg,
+		c:      c,
 	}
 }
 
 func (t *Periodic) Run() {
 	t.ctx, t.cancel = context.WithCancel(context.Background())
-	t.revs = make([]int64, 0)
 	clock := t.clock
+	timer := clock.After(t.period)
 
 	go func() {
-		last := clock.Now()
 		for {
-			t.revs = append(t.revs, t.rg.Rev())
 			select {
 			case <-t.ctx.Done():
 				return
-			case <-clock.After(checkCompactionInterval):
-				t.mu.Lock()
-				p := t.paused
-				t.mu.Unlock()
-				if p {
+			case <-timer:
+				timer = clock.After(t.period)
+				t.mu.RLock()
+				if t.paused {
+					t.mu.RUnlock()
 					continue
 				}
-			}
-
-			if clock.Now().Sub(last) < executeCompactionInterval {
-				continue
-			}
-
-			rev, remaining := t.getRev(t.periodInHour)
```

Comment:
it's not necessary to remove this to support the feature and removing it completely changes the behavior of the periodic compactor